### DISCUSSION TITLE
Interpreting VIMP for random forest classification

### DISCUSSION THREAD

**Discussion by DavidMarguerit** [kogalur/randomForestSRC#220] [Q&A] [answered]
First of all, thank you very much for your work. It is very useful. I am wondering how to interpret the feature importance for random forest classification. My response variable is binary (being a worker or not), and I would like to show which of my cofounders are the most important for being a worker. The command vimp() gives a table with three columns:
1) All
2) Being a worker
3) Not being a worker

It is unclear to me why the table has 3 columns and not one. Since my response is a binary variable, I would expect to have one column. For instance, if education is an important feature for being a worker, it should also be an important feature for not being a worker, since it is the opposite situation. However, the data shows that a feature could be important for being a worker but not important for not being a worker.

Do you have any document explaining how to interpret vimp() for random forest classification?

In my case, should I focus on column 1 or 2?

Thank you,
David

**Comment by ishwaran** [comment]
There are three columns provided because they represent VIMP for the unconditional and conditional case: "all" (unconditional), "being a worker" (conditional on the data where the class label is being a worker) and "not being a worker" (conditional on the data where the class label is not being a worker).  Almost always data analysts just focus on the first column "all".  However the conditional VIMP (columns 2 and 3) can sometimes provide surprising insight.

Also see specifying the type of predicted value ... from our getting started webpage:

https://luminwin.github.io/randomForestSRC/articles/getstarted.html#classification-1

**Reply by DavidMarguerit** [reply]
Thank you very much for your quick reply. It is very helpful.

Since in my project, I am focusing on predicting "being a worker" and that the event is rare (only 2.4% of individuals in my sample are workers), would this be an argument to focus on the accuracy of the "being a worker"?

I also notice that the columns "all" and "not being a worker" are pretty similar. I guess it is because most of the individuals are not workers. Am I right?

Thank you for being so helpful.
Best,
David

**Comment by ishwaran** [comment, marked as answer]
2.4% is a class imbalanced ratio of ~41 therefore I think you have a bigger problem.  Since the majority class label (not a worker) has such a high ratio compared to the minority class label (worker), you have what is called a class-imbalanced data set.

Therefore I recommend you use a class imbalanced analysis using the random forests quantile classifier (RFQ) implemented by the function `imbalanced`.  Then you should also use the custom made VIMP based on the gmean implemented by that solution.  You can read more about this on our [RFQ vignette](https://luminwin.github.io/randomForestSRC/articles/imbalance.html) and the [imbalanced helpfile](https://luminwin.github.io/randomForestSRC/reference/imbalanced.rfsrc.html).

**Comment by DavidMarguerit** [comment]
The `imbalanced` function works like a charm. Thank you! However, I would like to show the partial dependence plot for a specific variable, but it takes a lot of time to compute it. My code has been running for almost a week and I don't have the plot yet. My database is quite big (142,000 observations and 43 variables).

Any idea on how I can speed up the computation?

My code is : `plot.variable(my_rf, xvar.names = 'variable1', partial = TRUE)`

**Comment by ishwaran** [comment]
I suggest you use the fast interface `partial.rfsrc`.  The function `plot.variable` is just a user friendly wrapper that ultimately calls that function.  Then you can more specifically control things and it should be much faster.  Let us know.
